core/services/workflows/delegate.go

```
@@ -80,6 +91,12 @@ func (d *Delegate) ServicesForSpec(ctx context.Context, spec job.Job) ([]job.Ser
 	if err != nil {
 		return nil, err
 	}
+
+	err = d.engineRegistry.Add(spec.WorkflowSpec.WorkflowID, engine)
+	if err != nil {
+		d.logger.Errorf("delegate failed to register workflow engine for workflow name: %s id: %s: %v", cfg.WorkflowName.String(), cfg.WorkflowID, err)
```

? why aren't we returning the error? it if failed to add can we really continue?

---

This approach is backwards compatible - if we failed to add it to the engineRegistry the previous functionality is still locked in.

---

We should return an error here IMO -- this should really never happen and if it does this op should fail